src/System.Linq.Parallel/tests/WithCancellationTests.cs

```diff
@@ -157,5 +167,19 @@ public static void DontDoWorkIfTokenAlreadyCanceled()
 
             Assert.NotNull(oce);
         }
+
+        private static void DisposedEnumerator(ParallelQuery<int> query, bool delay = false)
+        {
+            query = query.WithCancellation(new CancellationTokenSource().Token).Select(x => x);
+
+            IEnumerator<int> enumerator = query.GetEnumerator();
+
+            enumerator.MoveNext();
+            if (delay) Task.Delay(1000).Wait();
```

Comment:
Yeah.
Part of the problem is many of the OuterLoop tests are using pretty large sets (1~4mil).  We'd probably need to cut some of that down, but it might be the only way to guarantee that some partitioning options are hit, or to exercise load balancing/etc.